doc/internals.rst

```diff
@@ -231,3 +231,450 @@ re-open it directly with Zarr:
+
+Xarray coders all have the same interface. They have two methods: ``decode``
+and ``encode``. The method ``decode`` takes a ``Variable`` in on-disk
+format and returns a ``Variable`` in Xarray format. Variable
+attributes no more applicable after the decoding, are dropped and stored in the
+``Variable.encoding`` to make them available to the ``encode`` method, which
+performs the inverse transformation.
+
+In the following an example on how to use the coders ``decode`` method:
+
+.. ipython:: python
+
+    var = xr.Variable(
+        dims=("x",), data=np.arange(10.0), attrs={"scale_factor": 10, "add_offset": 2}
+    )
+    var
+
+    coder = xr.coding.variables.CFScaleOffsetCoder()
+    decoded_var = coder.decode(var)
+    decoded_var
+    decoded_var.encoding
+
+Some of the transformations can be common to more backends, so before
+implementing a new decoder, be sure Xarray does not already implement that one.
+
+The backends can reuse Xarray’s decoders, either instantiating the coders
+and using the method ``decode`` directly or using the higher-level function
+:py:func:`~xarray.conventions.decode_cf_variables` that groups Xarray decoders.
+
+In some cases, the transformation to apply strongly depends on the on-disk
+data format. Therefore, you may need to implement your own decoder.
+
+An example of such a case is when you have to deal with the time format of a
+grib file. grib format is very different from the NetCDF one: in grib, the
+time is stored in two attributes dataDate and dataTime as strings. Therefore,
+it is not possible to reuse the Xarray time decoder, and implementing a new
+one is mandatory.
+
+Decoders can be activated or deactivated using the boolean keywords of
+Xarray :py:meth:`~xarray.open_dataset` signature: ``mask_and_scale``,
+``decode_times``, ``decode_timedelta``, ``use_cftime``,
+``concat_characters``, ``decode_coords``.
+Such keywords are passed to the backend only if the User sets a value
+different from ``None``.  Note that the backend does not necessarily have to
+implement all the decoders, but it shall declare in its ``open_dataset``
+interface only the boolean keywords related to the supported decoders.
+
+.. _RST backend_registration:
+
+How to register a backend
++++++++++++++++++++++++++++
+
+Define a new entrypoint in your ``setup.py`` (or ``setup.cfg``) with:
+
+- group: ``xarray.backend``
+- name: the name to be passed to :py:meth:`~xarray.open_dataset`  as ``engine``
+- object reference: the reference of the class that you have implemented.
+
+You can declare the entrypoint in ``setup.py`` using the following syntax:
+
+.. code-block::
+
+    setuptools.setup(
+        entry_points={
+            "xarray.backends": [
+                "engine_name=your_package.your_module:your_backendentrypoint"
+            ],
+        },
+    )
+
+in ``setup.cfg``:
+
+.. code-block:: cfg
+
+    [options.entry_points]
+    xarray.backends =
+        engine_name = your_package.your_module:your_backendentrypoint
```

Comment:
```suggestion
                "engine_name=your_package.your_module:YourBackendEntryClass"
            ],
        },
    )

in ``setup.cfg``:

.. code-block:: cfg

    [options.entry_points]
    xarray.backends =
        engine_name = your_package.your_module:YourBackendEntryClass
```

The above suggestion is a small naming change to emphasize the entrypoint object is a class.

Reply:
Thank you! You are right!